core/win32/inject.c

```diff
@@ -1185,7 +1189,10 @@ inject_gencode_mapped_helper(HANDLE phandle, char *dynamo_path, uint64 hook_loca
     byte *mode_switch_buf = NULL;
     byte *mode_switch_data = NULL;
     size_t switch_code_sz = PAGE_SIZE;
-    static const size_t switch_data_sz = SWITCH_MODE_DATA_SIZE;
+    size_t switch_data_sz = SWITCH_MODE_DATA_SIZE;
+    if (x86_code && DYNAMO_OPTION(inject_x64)) {
+	switch_data_sz *= 3; // we need space for saving ESP and EAX
```

Comment:
Not following why 3x?  Just +4?

Reply:
Fixed with +4 and proper name on last commit.